### PR TITLE
Updated CLI docs for compilation process

### LINKED ISSUE #2189: Update CLI docs involving magento setup:upgrade and setup:di:compile
Type of Doc:
- [ ] New topic
- [x] Topic Update, Devdoc link:
- [ ] Other

Magento 2 GitHub Issue: https://github.com/magento/magento2/issues/11044

Magento 2 GitHub PR: https://github.com/magento/magento2/pull/11047
Magento 2 version:
- [ ] 2.1.X
- [x] 2.2.X
- [x] 2.3.X
- [ ] back port
- [ ] forward port

Notifications of Changes:
- [ ] Changes to Admin interface?
- [ ] New software requirements?
- [ ] New REST endpoints or updated GraphQL query syntax?
- [x] New command line option?

## Content:
As a result of [PR-11044](https://github.com/magento/magento2/issues/11044), the `magento setup:upgrade` command no longer prompts to run `setup:di:compile` if Magento is in developer mode. 

There are several places in devdocs that state to run `setup:di:compile` after upgrading. Investigate whether the compile steps should be removed, or clarify whether the procedures are taking place while Magento is in default or production mode.

## Additional information:

Some possible topics to be updated:

https://devdocs.magento.com/guides/v2.2/comp-mgr/install-extensions.html
https://devdocs.magento.com/guides/v2.2/comp-mgr/install-extensions/b2b-installation.html
https://devdocs.magento.com/guides/v2.2/get-started/create-integration.html

### PR DESCRIPTION
**Pull request by gelanivishal**
## This PR is a:

- [ ] New topic
- [X] Content update
- [X] Content fix or rewrite
- [X] Bug fix or improvement

## Summary
I have added that the `magento setup:upgrade` command no longer prompts to run `setup:di:compile` if Magento is in developer mode
When this pull request is merged, it will fixed #2189 

## Additional information

List all affected URLs 

- https://devdocs.magento.com/guides/v2.2/comp-mgr/install-extensions.html
- https://devdocs.magento.com/guides/v2.2/comp-mgr/install-extensions/b2b-installation.html
- https://devdocs.magento.com/guides/v2.2/get-started/create-integration.html

whatsnew
Added information for the `setup:di:compile` step for [Install extensions from the command line](https://devdocs.magento.com/guides/v2.2/comp-mgr/install-extensions.html), [Install the B2B extension](https://devdocs.magento.com/guides/v2.2/comp-mgr/install-extensions/b2b-installation.html), and [Create an integration](https://devdocs.magento.com/guides/v2.2/get-started/create-integration.html).

### CODE REVIEW

**Comment by magento-cicd2** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.1/get-started/create-integration.md [hunk 1]
```diff
@@ -240,6 +240,8 @@ Use the following steps to install your module:
     <code>bin/magento setup:upgrade</code>
 
 2. Run the following command to generate the new code.
+{: .bs-callout .bs-callout-info }
+If you get the 'Please rerun Magento compile command' message then run below `bin/magento setup:di:compile` command. This message is only shows up in production mode.
```
Comment (lorikrell) [inline guides/v2.1/get-started/create-integration.md]: Recommended edit:

```suggestion
In Production mode, you may receive a message to 'Please rerun Magento compile command'.  Enter the command below. Magento does not prompt you to run the compile command in Developer mode.
```

File: guides/v2.2/comp-mgr/install-extensions/b2b-installation.md [hunk 2]
```diff
@@ -31,6 +31,8 @@ The {{site.data.var.b2b}} extension is only available for {{site.data.var.ee}} v
 2.  When prompted, enter your [authentication keys]({{ page.baseurl }}/install-gde/prereq/connect-auth.html). Your *public key* is your username; your *private key* is your password. If you have stored your public and private keys in `auth.json`, you won't be asked to enter them here.
 
 3.  Run the following commands after Composer finishes updating modules:
+{: .bs-callout .bs-callout-info }
```
Comment (lorikrell) [inline guides/v2.2/comp-mgr/install-extensions/b2b-installation.md]: Recommend entering a blank line above.

```suggestion

{: .bs-callout .bs-callout-info }
```

File: guides/v2.2/comp-mgr/install-extensions/b2b-installation.md [hunk 3]
```diff
@@ -31,6 +31,8 @@ The {{site.data.var.b2b}} extension is only available for {{site.data.var.ee}} v
...
+If you get the 'Please rerun Magento compile command' message then run below `bin/magento setup:di:compile` command. This message is only shows up in production mode.
```
Comment (lorikrell) [inline guides/v2.2/comp-mgr/install-extensions/b2b-installation.md]: Recommended edit:

```suggestion
In Production mode, you may receive a message to 'Please rerun Magento compile command'.  Enter the command below. Magento does not prompt you to run the compile command in Developer mode.
```

File: guides/v2.2/get-started/create-integration.md [hunk 4]
```diff
@@ -278,6 +278,8 @@ Use the following steps to install your module:
     <code>bin/magento setup:upgrade</code>
 
 2. Run the following command to generate the new code.
+{: .bs-callout .bs-callout-info }
```
Comment (lorikrell) [inline guides/v2.2/get-started/create-integration.md]: Recommend entering a blank line above.

```suggestion

{: .bs-callout .bs-callout-info }
```

File: guides/v2.2/get-started/create-integration.md [hunk 5]
```diff
@@ -278,6 +278,8 @@ Use the following steps to install your module:
...
+If you get the 'Please rerun Magento compile command' message then run below `bin/magento setup:di:compile` command. This message is only shows up in production mode.
```
Comment (lorikrell) [inline guides/v2.2/get-started/create-integration.md]: Recommended edit:

```suggestion
In Production mode, you may receive a message to 'Please rerun Magento compile command'.  Enter the command below. Magento does not prompt you to run the compile command in Developer mode.
```

File: guides/v2.3/comp-mgr/install-extensions/b2b-installation.md [hunk 6]
```diff
@@ -31,6 +31,8 @@ The {{site.data.var.b2b}} extension is only available for {{site.data.var.ee}} v
 2.  When prompted, enter your [authentication keys]({{ page.baseurl }}/install-gde/prereq/connect-auth.html). Your *public key* is your username; your *private key* is your password. If you have stored your public and private keys in `auth.json`, you won't be asked to enter them here.
 
 3.  Run the following commands after Composer finishes updating modules:
+{: .bs-callout .bs-callout-info }
```
Comment (lorikrell) [inline guides/v2.3/comp-mgr/install-extensions/b2b-installation.md]: Recommend entering a blank line:

```suggestion

{: .bs-callout .bs-callout-info }
```

File: guides/v2.3/comp-mgr/install-extensions/b2b-installation.md [hunk 7]
```diff
@@ -31,6 +31,8 @@ The {{site.data.var.b2b}} extension is only available for {{site.data.var.ee}} v
...
+If you get the 'Please rerun Magento compile command' message then run below `bin/magento setup:di:compile` command. This message is only shows up in production mode.
```
Comment (lorikrell) [inline guides/v2.3/comp-mgr/install-extensions/b2b-installation.md]: Recommended edit:

```suggestion
In Production mode, you may receive a message to 'Please rerun Magento compile command'.  Enter the command below. Magento does not prompt you to run the compile command in Developer mode.
```

File: guides/v2.3/get-started/create-integration.md [hunk 8]
```diff
@@ -271,6 +271,8 @@ Use the following steps to install your module:
     <code>bin/magento setup:upgrade</code>
 
 2. Run the following command to generate the new code.
+{: .bs-callout .bs-callout-info }
```
Comment (lorikrell) [inline guides/v2.3/get-started/create-integration.md]: Recommend entering a blank line above:

```suggestion

{: .bs-callout .bs-callout-info }
```

File: guides/v2.3/get-started/create-integration.md [hunk 9]
```diff
@@ -271,6 +271,8 @@ Use the following steps to install your module:
...
+If you get the 'Please rerun Magento compile command' message then run below `bin/magento setup:di:compile` command. This message is only shows up in production mode.
```
Comment (lorikrell) [inline guides/v2.3/get-started/create-integration.md]: Recommended edit:

```suggestion
In Production mode, you may receive a message to 'Please rerun Magento compile command'.  Enter the command below. Magento does not prompt you to run the compile command in Developer mode.
```

**Comment by lorikrell** [comment]
running tests

**Pull request merged**